pkg/istio-agent/xds_proxy.go

```diff
@@ -180,6 +179,10 @@ func (p *XdsProxy) UnregisterStream() {
 func (p *XdsProxy) RegisterStream(c *ProxyConnection) {
 	p.connectedMutex.Lock()
 	defer p.connectedMutex.Unlock()
+	if p.connected != nil {
```

Comment:
If this could not happen in real word, why do not fix the test by waiting until the stream is unregistered?

Reply:
it can happen, u just doubt it does often. If you recall from the workload entry discussion, we may register two connects before we register the initial disconnect. We should be robust against these cases

Reply:
Yeah, that's the concern